src/Bridge/Symfony/HttpKernel/Controller/ArgumentResolver/EnumValueResolver.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * This file is part of the "elao/enum" package.
+ *
+ * Copyright (C) Elao
+ *
+ * @author Elao <contact@elao.com>
+ */
+
+namespace Elao\Enum\Bridge\Symfony\HttpKernel\Controller\ArgumentResolver;
+
+use Elao\Enum\EnumInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Controller\ArgumentValueResolverInterface;
+use Symfony\Component\HttpKernel\ControllerMetadata\ArgumentMetadata;
+use Symfony\Component\HttpKernel\Exception\BadRequestHttpException;
+
+class EnumValueResolver implements ArgumentValueResolverInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function supports(Request $request, ArgumentMetadata $argument)
+    {
+        return is_subclass_of($argument->getType(), EnumInterface::class);
```

Comment:
`!$argument->isVariadic()` if we don't want to support variadics. Otherwise the resolve method should also handle arrays.

We must also check if the `$argument->getName()` is part of the attributes, query or request bags (using `has` on each).

Reply:
What do you mean? Why do we need those has checks?

Reply:
See `RequestAttributeValueResolver` for instance. The support method should return `true` only if the argument can be found in the request. If it isn't, the default value provided on the controller will be used (if provided, otherwise an exception).